libraries/mbed/targets/hal/TARGET_NORDIC/TARGET_MCU_NRF51822/serial_api.c

```diff
@@ -29,7 +29,7 @@
 
 static uint32_t serial_irq_ids[UART_NUM] = {0};
 static uart_irq_handler irq_handler;
-static uint32_t acceptedSpeeds[17][2] = {{1200, UART_BAUDRATE_BAUDRATE_Baud1200},
+static int acceptedSpeeds[17][2] = {{1200, UART_BAUDRATE_BAUDRATE_Baud1200},
```

Comment:
The mbed API uses an `int` for the function call where `acceptedSpeeds` is used, and this triggers a "signed and unsigned" comparison warning.

Since you probably don't want to change the API, it was easier to change this, especially since the `acceptedSpeeds` table is only accessed in one location.